src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Windows/Disk/WindowsDiskMetrics.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using Microsoft.Extensions.Options;
+using Microsoft.Shared.Instruments;
+
+#if NETCOREAPP
+using System.Runtime.Versioning;
+#endif
+
+namespace Microsoft.Extensions.Diagnostics.ResourceMonitoring.Windows.Disk;
+
+#if NETCOREAPP
+[SupportedOSPlatform("windows")]
+#endif
+internal sealed class WindowsDiskMetrics
+{
+    private const string LogicalDiskCategory = "LogicalDisk";
+
+    private const string DeviceKey = "system.device";
+
+    private const string DirectionKey = "disk.io.direction";
+
+    private static readonly KeyValuePair<string, object?> _directionReadTag = new(DirectionKey, "read");
+
+    private static readonly KeyValuePair<string, object?> _directionWriteTag = new(DirectionKey, "write");
+
+    private readonly Dictionary<string, WindowsDiskPerSecondPerfCounters> _perSecondCounters = new();
```

Comment:
[nit] Please avoid excessive spacing

```suggestion
    private const string LogicalDiskCategory = "LogicalDisk";
    private const string DeviceKey = "system.device";
    private const string DirectionKey = "disk.io.direction";

    private static readonly KeyValuePair<string, object?> _directionReadTag = new(DirectionKey, "read");
    private static readonly KeyValuePair<string, object?> _directionWriteTag = new(DirectionKey, "write");
    private readonly Dictionary<string, WindowsDiskPerSecondPerfCounters> _perSecondCounters = new();
```

Ditto in other files.